test/trento/clusters_test.exs

```diff
@@ -246,6 +246,55 @@ defmodule Trento.ClustersTest do
       [%ClusterReadModel{id: ^cluster_id, cib_last_written: ^cib_last_written}] =
         Clusters.get_all_clusters()
     end
+
+    test "should return clusters with enriched details" do
+      cluster_id = Faker.UUID.v4()
+
+      %{
+        name: node_name
+      } = node = build(:hana_cluster_node, attributes: %{foo_attribute: "foo_value"})
+
+      details = build(:hana_cluster_details, nodes: [node])
+
+      insert(:cluster_enrichment_data,
+        cluster_id: cluster_id,
+        details: %{
+          nodes: [
+            %{
+              name: node_name,
+              attributes: %{
+                bar_attribute: "bar_value"
+              }
+            }
+          ]
+        }
+      )
+
+      insert(:cluster, id: cluster_id, details: details)
+
+      expected_details =
+        %{
+          details
+          | nodes: [
+              %{
+                node
+                | attributes: %{
+                    "foo_attribute" => "foo_value",
+                    "bar_attribute" => "bar_value"
+                  }
+              }
+            ]
+        }
+        |> Jason.encode!()
+        |> Jason.decode!()
```

Comment:
duped entry.
in any case, i don't know why we need to encode to json, as the details are a struct

Reply:
unfortunately necessary to match what we get from the db with what we have at hand.
when we do insert of factories we have shiny Structs, when we query back from the db the inner detail is just a plain map with "key" => "value" instead of key: "value" :shrug: